### PR TITLE
fix(aws-sdk): use RPC attributes from spec

### LINKED ISSUE #5159: aws-sdk instrumentation is missing the RPC attributes from specification
**Describe the bug**
According to the [spec for Semantic conventions for AWS SDK](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md#semantic-conventions-for-aws-sdk), the following attributes are expected for aws-sdk spans: `rpc.method`, `rpc.service`, `rpc.system`.

**Steps to reproduce**
Just run the instrumentation, or browse the code.

**What did you expect to see?**
I expected these attributes to be published by the aws-sdk 1.11 and aws-sdk 2.2 instrumentation according to the spec.

**What did you see instead?**
These attributes are published under different, non-spec attributes which are available only in experimental mode: `aws.operation` and `aws.service`

**What version are you using?**
latest `1.10.0`

**Additional context**
I want to suggest renaming these attributes so they are spec-compliant and to always report them (not only when `otel.instrumentation.aws-sdk.experimental-span-attributes` is true).
If it makes sense, I can work on PR to do that.

### PR DESCRIPTION
**Pull request by blumamir**
Fix #5159 

For `aws-sdk-1.11` and `aws-sdk-2.2` instrumentations:
- Rename `aws.service` to `rpc.service` [according to the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md#common-attributes).
- Rename `aws.operation` to `rpc.method` [according to the spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md#common-attributes).
- Add missing attribute [from spec](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/instrumentation/aws-sdk.md#common-attributes): `rpc.system`
- The above rpc attributes are always published, unlike the `aws.*` attributes which are set only when `otel.instrumentation.aws-sdk.experimental-span-attributes=true`
- Renamed `AwsSdkAttributesExtractor.java` to `AwsSdkHttpAttributesExtractor.java`, to make it consistent with the other extractors in the module. This extractor is extracting http client attributes only, and not aws-sdk domain attributes.
- Fixed the tests to use the new attributes and removed the old non-spec-compliant names

This is my first PR in java, so please take a good look.

### CODE REVIEW

**Comment by blumamir** [comment]
A test seems to fail in CI (`PR build / test (15, openj9) (pull_request) Failing after 29m — test (15, openj9)`), but I don't think it's related to my changes:
```
 Task :instrumentation:spring:spring-rmi-4.0:javaagent:test

SpringRmiTest > Client call creates spans FAILED
    java.rmi.ConnectException: Connection refused to host: 10.1.0.130; nested exception is: 
    	java.net.ConnectException: Connection timed out
        at java.rmi@15.0.2/sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:623)
        at java.rmi@15.0.2/sun.rmi.transport.tcp.TCPChannel.createConnection(TCPChannel.java:209)
        at java.rmi@15.0.2/sun.rmi.transport.tcp.TCPChannel.newConnection(TCPChannel.java:196)
        at java.rmi@15.0.2/sun.rmi.server.UnicastRef.invoke(UnicastRef.java:132)
        at java.rmi@15.0.2/java.rmi.server.RemoteObjectInvocationHandler.invokeRemoteMethod(RemoteObjectInvocationHandler.java:217)
        at java.rmi@15.0.2/java.rmi.server.RemoteObjectInvocationHandler.invoke(RemoteObjectInvocationHandler.java:162)
        at app//org.springframework.remoting.rmi.RmiClientInterceptor.doInvoke(RmiClientInterceptor.java:399)
        at app//org.springframework.remoting.rmi.RmiClientInterceptor.doInvoke(RmiClientInterceptor.java:345)
        at app//org.springframework.remoting.rmi.RmiClientInterceptor.invoke(RmiClientInterceptor.java:260)
        at app//org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:179)
        at app//org.springframework.aop.framework.JdkDynamicAopProxy.invoke(JdkDynamicAopProxy.java:207)
        at SpringRmiTest.Client call creates spans_closure1(SpringRmiTest.groovy:72)
        at SpringRmiTest.Client call creates spans_closure1(SpringRmiTest.groovy)
        at app//groovy.lang.Closure.call(Closure.java:412)
        at app//io.opentelemetry.instrumentation.testing.TestInstrumenters.runWithInstrumenter(TestInstrumenters.java:84)
        at app//io.opentelemetry.instrumentation.testing.TestInstrumenters.runWithSpan(TestInstrumenters.java:44)
        at app//io.opentelemetry.instrumentation.testing.InstrumentationTestRunner.runWithSpan(InstrumentationTestRunner.java:119)
        at app//io.opentelemetry.instrumentation.test.InstrumentationSpecification.runWithSpan(InstrumentationSpecification.groovy:120)
        at SpringRmiTest.Client call creates spans(SpringRmiTest.groovy:72)

        Caused by:
        java.net.ConnectException: Connection timed out
            at java.base@15.0.2/sun.nio.ch.Net.connect(Net.java:574)
            at java.base@15.0.2/sun.nio.ch.Net.connect(Net.java:563)
            at java.base@15.0.2/sun.nio.ch.NioSocketImpl.connect(NioSocketImpl.java:588)
            at java.base@15.0.2/java.net.SocksSocketImpl.connect(SocksSocketImpl.java:333)
            at java.base@15.0.2/java.net.Socket.connect(Socket.java:648)
            at java.base@15.0.2/java.net.Socket.connect(Socket.java:597)
            at java.base@15.0.2/java.net.Socket.<init>(Socket.java:520)
            at java.base@15.0.2/java.net.Socket.<init>(Socket.java:294)
            at java.rmi@15.0.2/sun.rmi.transport.tcp.TCPDirectSocketFactory.createSocket(TCPDirectSocketFactory.java:40)
            at java.rmi@15.0.2/sun.rmi.transport.tcp.TCPEndpoint.newSocket(TCPEndpoint.java:617)
            ... 18 more
```

Is it just flacky test? should I ignore it?

**Comment by mateuszrzeszutek** [comment]
> Is it just flacky test? should I ignore it?

Yes, that's just a flaky test that's been introduced very recently - don't worry about it. I'll rerun the checks on your branch just in case.

File: instrumentation/aws-sdk/aws-sdk-1.11/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/AwsSdkRpcAttributesExtractor.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v1_11;
+
+import com.amazonaws.Request;
+import com.amazonaws.Response;
+import io.opentelemetry.instrumentation.api.instrumenter.rpc.RpcAttributesExtractor;
+
+class AwsSdkRpcAttributesExtractor extends RpcAttributesExtractor<Request<?>, Response<?>> {
```
Comment (anuraaga) [inline instrumentation/aws-sdk/aws-sdk-1.11/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v1_11/AwsSdkRpcAttributesExtractor.java:12]:
```suggestion
final class AwsSdkRpcAttributesExtractor extends RpcAttributesExtractor<Request<?>, Response<?>> {
```

File: instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/AwsSdkRpcAttributesExtractor.java [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+ */
+
+package io.opentelemetry.instrumentation.awssdk.v2_2;
+
+import io.opentelemetry.instrumentation.api.instrumenter.rpc.RpcAttributesExtractor;
+import software.amazon.awssdk.core.interceptor.ExecutionAttributes;
+import software.amazon.awssdk.core.interceptor.SdkExecutionAttribute;
+import software.amazon.awssdk.http.SdkHttpResponse;
+
+class AwsSdkRpcAttributesExtractor
```
Comment (anuraaga) [inline instrumentation/aws-sdk/aws-sdk-2.2/library/src/main/java/io/opentelemetry/instrumentation/awssdk/v2_2/AwsSdkRpcAttributesExtractor.java:13]:
```suggestion
final class AwsSdkRpcAttributesExtractor
```

**Review by anuraaga — APPROVED** [review]
Thanks1

**Review by mateuszrzeszutek — APPROVED** [review]

**Review by laurit — APPROVED** [review]

**Pull request merged**